paymentsheet/src/main/java/com/stripe/android/lpmfoundations/paymentmethod/PaymentMethodMetadata.kt

```diff
@@ -197,6 +197,34 @@ internal data class PaymentMethodMetadata(
         }
     }
 
+    fun allowRedisplay(saveForFutureUse: Boolean?): PaymentMethod.AllowRedisplay {
+        return when (paymentMethodSaveConsentBehavior) {
+            PaymentMethodSaveConsentBehavior.Legacy -> PaymentMethod.AllowRedisplay.UNSPECIFIED
+            PaymentMethodSaveConsentBehavior.Disabled -> {
+                if (hasIntentToSetup()) {
+                    PaymentMethod.AllowRedisplay.LIMITED
+                } else {
+                    PaymentMethod.AllowRedisplay.UNSPECIFIED
+                }
+            }
+            PaymentMethodSaveConsentBehavior.Enabled -> {
+                if (hasIntentToSetup()) {
+                    if (saveForFutureUse == true) {
+                        PaymentMethod.AllowRedisplay.ALWAYS
+                    } else {
+                        PaymentMethod.AllowRedisplay.LIMITED
```

Comment:
`SetupIntent`s requires that the payment method be available at some level to charge it when needed (subscriptions for example). We at minimum need to set the `redisplay` value to `Limited` for payment methods being used to confirm `SetupIntent`s.